app/controllers/users.rb

```diff
@@ -0,0 +1,36 @@
+# get a list of users
+get '/users' do
+end
+# get a form to add a new users
+get '/users/new' do
+
+  erb :new_user
+end
+# add a new users
+post '/users' do
+  session[:errors] = nil
+  user = User.create(username: params[:username], email: params[:email], password: params[:password])
+  if user.errors.any?
+    session[:errors] = user.display_errors
+    redirect '/users/new' #<< partial (: invalid_user )
+  elsif
+    session[:user_id] = user.id
+    session[:user_password] = user.password
+    redirect '/'
+  end
+end
+# get a specific instance of users
+get '/users/:username' do
+  @user = User.find_by(username:params[:username])
+  @shoes_with_reviews = @user.shoes_with_reviews(@user.id)
```

Comment:
`@user.shoes_with_reviews(@user.id)` seems weird because `shoes_with_reviews` is a method off of a `User` instance and should therefore know the user's id and not need you to pass in `(@user.id)`